### PR TITLE
Category Refactor

### LINKED ISSUE #972: Category Refactor
- [x] use local store

### PR DESCRIPTION
**Pull request by SteRiccio**
Resolves #972

- [x] Creation and fetch
- [x] Category props update (name)
- [x] Import from CSV
- [x] Levels edit
- [x] Items edit

### CODE REVIEW

File: webapp/components/survey/CategoryDetails/ImportSummary/TableRow.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+
```
Comment (minotogna) [inline webapp/components/survey/CategoryDetails/ImportSummary/TableRow.js:3]: even this to its own folder

File: webapp/components/survey/CategoryDetails/LevelDetails/LevelDetails.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+
+import * as Category from '@core/survey/category'
+import * as CategoryLevel from '@core/survey/categoryLevel'
+import * as CategoryItem from '@core/survey/categoryItem'
+import * as Validation from '@core/validation/validation'
+import { normalizeName } from '@core/stringUtils'
```
Comment (minotogna) [inline webapp/components/survey/CategoryDetails/LevelDetails/LevelDetails.js:8]: better full import. Import * as StringUtils

File: webapp/service/api/categories/index.js [hunk 3]
```diff
@@ -9,10 +9,10 @@ export const fetchCategories = async ({ surveyId, draft = true, validate = false
   return categories
 }
 
-export const fetchCategory = async ({ surveyId, Uuid, draft = true, validate = true }) => {
+export const fetchCategory = async ({ surveyId, uuid, draft = true, validate = true }) => {
```
Comment (minotogna) [inline webapp/service/api/categories/index.js:12]: use categoryUuid from master

File: webapp/components/survey/CategoryDetails/ImportSummary/TableHeader.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import React from 'react'
+
```
[the remark this answers is not in the document]
Comment (SteRiccio) [reply webapp/components/survey/CategoryDetails/ImportSummary/TableHeader.js:2]: ok

File: webapp/components/survey/CategoryDetails/ImportSummary/TableRow.js (hunk 1 above)
Comment (SteRiccio) [reply webapp/components/survey/CategoryDetails/ImportSummary/TableRow.js:3]: ok

File: webapp/components/survey/CategoryDetails/LevelDetails/LevelDetails.js (hunk 2 above)
Comment (SteRiccio) [reply webapp/components/survey/CategoryDetails/LevelDetails/LevelDetails.js:8]: ok

File: webapp/service/api/categories/index.js (hunk 3 above)
Comment (SteRiccio) [reply webapp/service/api/categories/index.js:12]: ok

File: webapp/components/survey/CategoryDetails/store/actions/category/useInit.js [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+  return category
+}
+
+const _fetchCategory = async ({ surveyId, categoryUuid }) => {
+  const {
+    data: { category },
+  } = await axios.get(`/api/survey/${surveyId}/categories/${categoryUuid}?draft=true&validate=true`)
+
+  return category
+}
```
Comment (ramirobg94) [inline webapp/components/survey/CategoryDetails/store/actions/category/useInit.js:32]: now this is on Service/api

Comment (SteRiccio) [reply webapp/components/survey/CategoryDetails/store/actions/category/useInit.js:32]: this file needs to be updated when the new ButtonMetaItemAdd will be used to create a new category... can you update this too in your pull request?

**Pull request merged**